### ISSUE TITLE
[BUG] Prompt.confirm `default`

### ISSUE THREAD

**Issue by ewels** [willmcgugan/rich#170]
**Describe the bug**
This isn't exactly a bug, but rather some unexpected behaviour with `Confirm.ask()`.

In some situations I would like to confirm something with the user, but set the default response to _Yes_ instead of _No_. I thought I'd just stick in a `default=True` argument to the function to see what happens, and it _kind of_ works 😅 Hitting enter gives the default response. The problem is that the default is shown in brackets but the `[y/N]` suffix remains unchanged, instead of being `[Y/n]`.

This isn't really a problem for me - I'll just not specify a default (I actually prefer requiring an explicit response, but I was converting from `click.confirm()` which _always_ has a default). I just thought I'd flag the behaviour.

Feel free to close if it's a feature and not a bug 😄 

**To Reproduce**

```python
from rich.prompt import Confirm
if Confirm.ask("Do you like rich?", default=True):
  print("Is the correct answer!")
else:
  print("Boo")
```

```console
$ python test.py
Do you like rich? [y/N] (True):
Is the correct answer!
```

**Platform**

* Rich 3.4.0
* iTerm2
* Mac OSX

**Comment by willmcgugan** [comment]
Ah. I copied this from click, and did wonder why it was capitalized that way, but it didn't occur to me the upper case letter was the default. Obvious now.

I don't find that terribly intuitive. Perhaps its because it's the first time I've seen that convention. How about leaving the choices lower case, but putting the default in text form in braces. Like this:

```
Do you like rich? [y/n] (y)
```

**Comment by ewels** [comment]
Works for me 👍🏻

**Comment by willmcgugan** [comment]
Fixed in v4.0.0

**Comment by ewels** [comment]
Brilliant stuff - thank you!
